climada/hazard/landslide.py

```diff
@@ -21,160 +21,99 @@
-            LOGGER.error("Start date must lie before end date. Please change")
-            raise ValueError
-
-        url = CONFIG.hazard.landslide.resources.opensearch.str()
-        params = dict(
-            q='global_landslide_nowcast',
-            limit=50,
-            startTime=starttime,
-            endTime=endtime)
-
-        resp = requests.get(url=url, params=params)
-        data = resp.json()
-        tif_url = []
-        for item in data['items']:
-            # extract json resonse snippet for tiff download
-            tif_url.append(item['action'][1]['using'][2]['url'])
-
-        resp_tif = []
-        for url in tif_url:
-            LOGGER.info('requesting %s', url)
-            resp_tif.append(requests.get(url=url))
-            LOGGER.info('downloading content...')
-            with Path(save_path, 'LS_nowcast_' + str(url[-12:-4]) + '.tif').open('wb') as fp:
-                fp.write(resp_tif[-1].content)
-
-    elif tif_type == "monthly":
-
-        command_line = 'curl -LO ' + CONFIG.hazard.landslide.resources.climatology_monthly.str()
-        args = shlex.split(command_line)
-        p = subprocess.Popen(args,
-                             stdout=subprocess.PIPE,
-                             stderr=subprocess.STDOUT)
-        p.communicate()
-
-
-def combine_nowcast_tiff(ls_folder_path, search_criteria='LS*.tif', operator="maximum"):
-    """Function to overlay several tiff files with landslide hazard data either by
-    keeping maximum value per pixel or by summing up all pixel values.
-    UPDATE: SOMETIMES WORKS, SOMETIMES NOT, ISSUE SEEMS TO BE WITH THE SHELL=TRUE COMMAND
-    Parameters:
-        ls_folder_path (str): folder path where landslide files are stored.
-        search_criteria (str): common name identifier of files that should be combined.
-            Either 'LS*.tif' for daily NASA nowcast maps or '*5400.tif' for monthly NASA maps.
-        operator (str): "maximum" keeps maximum value for each pixel throughout files,
-            "sum" sums up all values for each pixel throughout files.
-     Returns:
-        combined_nowcasts_LS.tif (tiff): 1 Tiff file combining all input tiffs.
+    if isinstance(res, tuple):
+        res_x, res_y = res
+    else:
+        res_x = res_y = res
+    col = int((geometry.x - xmin) / res_x)
+    row = int((ymax - geometry.y) / res_y)
+    return col, row
+    
+def mapping_grid2flattened(col, row, matrix_shape):
+    """ given a col and row index and the initial 2D matrix shape,
+    return the 1-dimensional index of the same point in the flattened matrix
+    - assumes concatenation in x-direction 
+    
+    Parameters
+    ----------
+    col : int
+        Column Index of an entry in the original matri
+    row : int
+        Row index of an entry in the original matrix
+    Returns
+    -------
+    index (1D) of the point in flattened array
     """
+    return row*matrix_shape[1]+col
+
+def sample_events_from_probs(prob_matrix, n_years):
```

Comment:
Also, would it not be useful to allow for different kinds of distributions besides the binomial?

Reply:
Yes, I agree. So, would it I) be useful to create a sampling util module? And II), do you think I should already include a kwarg for the type of distribution, in case more options will be implemented under a `sample_events_from_probs` parent function?

Reply:
1) Either a sampling util module, or maybe it would also fit in an engine module. Maybe we should a) open an issue on this for an open discussion and b) form a working sub-group at the developer's meeting.
2) I think this could be useful for this particular module, no? Maybe a poisson distribution could be appropriate for sampling the landslide events? In this case, I would say yes, adding a kwarg for the distribution is useful. However, if it is only for a possible future compatibility, I think we can leave it as is because we do not yet really know what the future implementation will look like. Right?

Reply:
as a general remark: I'll open an issue, and for now leave it in the landslide module with two distribution options (binom and poisson)

Reply:
I might have missed it, but I do not see the issue on this matter.